src/H5Gdeprec.c

```diff
@@ -519,10 +519,19 @@ H5Gmove2(hid_t src_loc_id, const char *src_name, hid_t dst_loc_id, const char *d
     H5VL_loc_params_t loc_params1;
     H5VL_object_t    *vol_obj2 = NULL; /* Object of dst_id */
     H5VL_loc_params_t loc_params2;
+    H5I_type_t        id_type   = H5I_BADID;
     herr_t            ret_value = SUCCEED; /* Return value */
 
     FUNC_ENTER_API(FAIL)
 
+    id_type = H5I_get_type(src_loc_id);
```

Comment:
You will need to handle allowing the magic H5L_SAME_LOC values for the src & dst IDs.

Reply:
Oh ok, hmm, so if the caller passes in 0, we assume they meant H5L_SAME_LOC?

Reply:
Also, the function has already handled allowing the magic H5L_SAME_LOC values, except that
loc_params1.obj_type                     = H5I_get_type(src_loc_id);  <- later causing segfault

happens before

    if (H5L_SAME_LOC != src_loc_id)
        /* get the location object */